server/build.gradle

```diff
@@ -353,17 +365,8 @@ tasks.named("missingJavadoc").configure {
    */
   dependsOn("generateProto")
   javadocMissingIgnore = [
-    "org.opensearch.extensions.proto.ExtensionIdentityProto",
-    "org.opensearch.extensions.proto.ExtensionIdentityProto.ExtensionIdentityOrBuilder",
-    "org.opensearch.extensions.proto.RegisterRestActionsProto",
-    "org.opensearch.extensions.proto.RegisterRestActionsProto.RegisterRestActionsOrBuilder",
-    "org.opensearch.extensions.proto.ExtensionRequestProto",
-    "org.opensearch.extensions.proto.ExtensionRequestProto.ExtensionRequestOrBuilder",
-    "org.opensearch.extensions.proto.RegisterTransportActionsProto",
-    "org.opensearch.extensions.proto.RegisterTransportActionsProto.RegisterTransportActionsOrBuilder",
-    "org.opensearch.extensions.proto.ExtensionTransportMessageProto",
-    "org.opensearch.extensions.proto.ExtensionTransportMessageProto.ExtensionTransportMessageOrBuilder",
-    "org.opensearch.extensions.proto"
+    "org.opensearch.extensions.proto",
```

Comment:
1/ Yes, we discussed this point here -> https://github.com/opensearch-project/OpenSearch/issues/7264. 

2/ This change won't allow arbitrary set of source files to be ignored.  You will see that my PR includes an ALLOW_LIST and it only contains "org.opensearch.extensions.proto" package.  I made this PR as a way to make progress on this issue.

3/ As I pointed out above, @Generated only seems to work for classes, but missingJavadoc checks are also performed on packages and interfaces.  Which is why I kept both the ALLOW_LIST based wildcards and the annotation in this PR.

Reply:
I get what you mean.  I think the assumption made in #7264 is that the package being ignored only contains generated source.  Then, it should be OK to restrict the change in this PR to only target generated source, unless we want to also support regex type grouping of source files to be ignored.  I just don't want that issue to linger around because some corner case is not covered.

Reply:
Thanks @austintlee , I thought `@Generated` would help (it did but not fully), we could help to address that

Reply:
(PS: this is just my opinion, I have no intention to block the change if other maintainers see value taking this path)

Reply:
Let's proceed with ignoring only the annotated classes.  This seems to satisfy the immediate need.

Reply:
I was able to find a way to cover the enclosed interfaces and the enclosing package.  Please take a look at the latest diff.  Thank you.